datamade/just-spaces
# Survey metadata

## Overview

Connects #63. Creates elements for survey metadata. Related issues that came out of this but should be addressed separately: #88; #91; #92; #93 

### To Do 

Add fields on survey model:

- [x] study

Add form elements:
- [x] total
- [x] time_start
- [x] time_stop
- [x] time_character
- [x] representation
- [x] microclimate
- [x] temperature_c
- [x] method

Other things:

- [x] We'll need observational surveys to require a `Total` form element. And maybe `time_start`, `time_stop`, `method`, and `representation`. Open a new issue to handle these. (update: issue #88)
- [x] Get data handler working with all of this
- [x] Figure out relationship between time_stop and submission time (update: issue #91)

## Testing Instructions

 * If you don't already have it, create the database: `pg_restore -C -j4 --no-owner just-spaces.dump | psql`
* Migrate: `python manage.py migrate`
* Login as testuser (password in the DataMade LastPass under "Just Spaces Staging")
* Create a new survey
* Add some of the metadata fields
* Publish your survey
* Submit it. Do the fields save as you'd expect on the survey model?

Connects #63
